akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -372,8 +384,17 @@ private[remote] class Decoder(
                     attemptsLeft - 1,
                     recipientPath, inboundEnvelope), retryResolveRemoteDeployedRecipientInterval)
                 else {
+                  // No more attempts left. If the retried resolve isn't successful the ref is banned and
+                  // we will not do the delayed retry resolve again. The reason for that is
+                  // if many messages are sent to such dead refs the resolve process will slow
+                  // down other messages.
+                  if (bannedRemoteDeployedActorRefs.size >= 100) {
```

Comment:
I don't think we want to write another LRU cache right now. I used the simplest thing that could solve the problem. The other LRU cache doesn't support this usage pattern.

This is at one place, and it is easy to understand that it is bounded by reading this code, so I don't understand your concern.